opensfm/commands/detect_features.py

```diff
@@ -51,42 +52,51 @@ def write_report(self, data, wall_time):
 
 
 def detect(args):
+    image, data = args
+
     log.setup()
 
-    image, data = args
+    if data.feature_index_exists(image):
+        logger.info('Skip recomputing {} features for image {}'.format(
+            data.feature_type().upper(), image))
+        return
+
     logger.info('Extracting {} features for image {}'.format(
         data.feature_type().upper(), image))
 
-    if not data.feature_index_exists(image):
-        start = timer()
-        mask = data.load_combined_mask(image)
-        if mask is not None:
-            logger.info('Found mask to apply for image {}'.format(image))
-        preemptive_max = data.config['preemptive_max']
-        p_unsorted, f_unsorted, c_unsorted = features.extract_features(
-            data.load_image(image), data.config, mask)
-        if len(p_unsorted) == 0:
-            return
-
-        size = p_unsorted[:, 2]
-        order = np.argsort(size)
-        p_sorted = p_unsorted[order, :]
-        f_sorted = f_unsorted[order, :]
-        c_sorted = c_unsorted[order, :]
-        p_pre = p_sorted[-preemptive_max:]
-        f_pre = f_sorted[-preemptive_max:]
-        data.save_features(image, p_sorted, f_sorted, c_sorted)
-        data.save_preemptive_features(image, p_pre, f_pre)
-
-        if data.config['matcher_type'] == 'FLANN':
-            index = features.build_flann_index(f_sorted, data.config)
-            data.save_feature_index(image, index)
-
-        end = timer()
-        report = {
-            "image": image,
-            "num_features": len(p_sorted),
-            "wall_time": end - start,
-        }
-        data.save_report(io.json_dumps(report),
-                         'features/{}.json'.format(image))
+    start = timer()
```

Comment:
We should skip computing the features if they are already computed.
Before we were using `data.feature_index_exists(image)` that will not work for words so we need something else.